core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -431,7 +433,7 @@ private static boolean checkPosOccurrenceParamValues(int position,
         throw RESOURCE.invalidIntegerInputForRegexpFunctions(Integer.toString(occurrencePosition),
             "occurrence_position", methodName).ex();
       }
-      if (position <= value.length()) {
+      if (position > value.length()) {
```

Comment:
This wasn't wrong before, I just inverted the condition to make the return boolean more intuitive wrt helper function name: validatePosOccurrenceParamValues(), now true if everything is valid and false otherwise. 
I've inverted the boolean checks for this method usages below, if you think this is more confusing we can revert to original.